File: src/algorithms/hash-maps/counting/first-unique-character/sources/FirstUniqueCharacter_test.cpp
#include "FirstUniqueCharacter.cpp"
#include <cassert>
#include <iostream>

int main() {
    assert(firstUniqueCharacter("leetcode") == 0);
    assert(firstUniqueCharacter("loveleetcode") == 2);
    assert(firstUniqueCharacter("aabb") == -1);
    assert(firstUniqueCharacter("z") == 0);
    assert(firstUniqueCharacter("aabbcc") == -1);
    assert(firstUniqueCharacter("aabc") == 2);
    assert(firstUniqueCharacter("abcde") == 0);
    assert(firstUniqueCharacter("abab") == -1);
    assert(firstUniqueCharacter("aadadaad") == -1);
    assert(firstUniqueCharacter("aba") == 1);

    std::cout << "All tests passed!" << std::endl;
    return 0;
}

File: src/algorithms/hash-maps/counting/first-unique-character/__tests__/FirstUniqueCharacter_test.java
public class FirstUniqueCharacter_test {
    public static void main(String[] args) {
        assert FirstUniqueCharacter.firstUniqueCharacter("leetcode") == 0;
        assert FirstUniqueCharacter.firstUniqueCharacter("loveleetcode") == 2;
        assert FirstUniqueCharacter.firstUniqueCharacter("aabb") == -1;
        assert FirstUniqueCharacter.firstUniqueCharacter("z") == 0;
        assert FirstUniqueCharacter.firstUniqueCharacter("aabbcc") == -1;
        assert FirstUniqueCharacter.firstUniqueCharacter("aabc") == 2;
        assert FirstUniqueCharacter.firstUniqueCharacter("abcde") == 0;
        assert FirstUniqueCharacter.firstUniqueCharacter("abab") == -1;
        assert FirstUniqueCharacter.firstUniqueCharacter("aadadaad") == -1;
        assert FirstUniqueCharacter.firstUniqueCharacter("aba") == 1;

        System.out.println("All tests passed!");
    }
}
